### PR TITLE
Argument order issue test

### PR DESCRIPTION
**Pull request by cryptix**
see #190

### CODE REVIEW

File: driver/weird_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package driver
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+)
+
+func check(err error, t *testing.T) {
```
Comment (cznic) [inline driver/weird_test.go]: I can reproduce the failure, but using this function hides where it actually happens. I can recover the information by eg. inserting a panic here, but please use `t.Fail*` directly at the failure place and remove this function, thank you.

File: driver/weird_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	var pos position
+	pos.Y = 0.1
+	pos.Z = 0.2
+	pos.Alpha = 0.3
+	pos.Beta = 0.4
+	pos.Comment = "small"
+
+	// prepare db
+	tx, err := db.Begin()
+	check(err, t)
```
Comment (cznic) [inline driver/weird_test.go]: I'm aware that replacing all this nice single line checks with at least 3 lines is a bit painful, but please use old fashioned

```go
if err != nil {
        t.Fatal(err)
}
```

everywhere, thank you.

File: driver/weird_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	defer stmt.Close()
+	res, err = stmt.Exec(pos.ID, pos.Comment, pos.Y, pos.Z, pos.Alpha, pos.Beta)
+	check(err, t)
+	cnt, err = res.RowsAffected()
+	check(err, t)
+	err = tx.Commit()
+	check(err, t)
+	if cnt != 1 {
+		fmt.Printf("affected: %d\n", cnt)
```
Comment (cznic) [inline driver/weird_test.go]: Please `s/fmt.Printf/t.Logf/`, assuming it's not an error actually.

**Comment by cryptix** [comment]
I removed the `check(err,t)` as requested and will try to move it out of the `driver` package. I just added it there since it was simple to copy my reproduction in there. I haven't used ql without the driver yet but looking at the examples in `all_test.go` it looks straight forward to use, just need to convert the statements.

> Glad to learn more about what you've possibly already figured out about this.

Not much more to add right now other than it feels  _off-by-one_-ish. `Exec(float, id, string)` also leads to the error for instance. I just had luck finding a combination that worked.

**Comment by cznic** [comment]
> I haven't used ql without the driver yet but looking at the examples in all_test.go it looks straight forward to use, just need to convert the statements.

Feel free to keep it where it's right now or move one directory up. I'd prefer the parent directory but it's not critical.

**Comment by cznic** [comment]
BTW, you surely _can_ use the driver from the parent directory. No need to rewrite the test.

**Comment by cznic** [comment]
Example: https://github.com/cznic/ql/blob/8c32ff10fdc2810e93a6e127119d95545497909e/all_test.go#L2005

**Comment by cryptix** [comment]
Feel free to merge #192. These 7 commits felt to noisy just for the reproduction of this bug.

**Pull request closed without merging**